src/node_jsvmapi.h

```diff
@@ -116,80 +116,99 @@ enum napi_valuetype {
   napi_function,
 };
 
-// Environment
-NODE_EXTERN napi_env napi_get_current_env();
+enum napi_errorvalue {
```

Comment:
Thinking about this more, I'm not sure a bit flag would be necessary, or worth the added complexity. If there is a status code like `napi_pending_exception`, then we can return that code when there is a pending exception and no other more-specific error conditions were detected.

If the return code is `napi_ok`, then it should be safe for the caller to assume there is no pending exception, assuming there is code in the NAPI implementation to check for a pending exception before returning.

If the return code is some error other than `napi_pending_exception`, then I think the caller cannot make any assumption about whether there is or is not a pending exception; they'll just have to call `napi_is_exception_pending()`. Is that OK, or should we try to make some sort of guarantee?

Checking for a pending exception *from within NAPI code* should be very fast: both the retrieval of V8's static `lastException` persistent value and the `IsEmpty()` check on it should be inlined. (But that won't be the case when calling across the NAPI boundary.)